homeassistant/components/varta_storage/sensor.py

```diff
@@ -0,0 +1,72 @@
+"""Sensor platform of the VARTA Storage integration."""
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Initialize the integration."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        VartaStorageEntity(coordinator, idx) for idx, ent in enumerate(coordinator.data)
+    )
+
+
+class VartaStorageEntity(CoordinatorEntity, SensorEntity):
+    """An entity using CoordinatorEntity.
+
+    The CoordinatorEntity class provides:
+      should_poll
+      async_update
+      async_added_to_hass
+      available
+
+    """
+
+    def __init__(self, coordinator, idx):
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self.idx = idx
+
+        self._attr_device_info = DeviceInfo(
+            configuration_url="http://" + coordinator.config_entry.data["host"],
+            identifiers={(DOMAIN, str(coordinator.config_entry.unique_id))},
```

Comment:
The device has a serial number right? Why not use that?

Reply:
The unique_id contains the serial number